client/src/components/Header/SearchBar/index.tsx

```diff
@@ -0,0 +1,201 @@
+      const { target } = e;
+      if (!(target instanceof HTMLElement)) return;
+      if (!target.closest('#search')) setShowWordList(false);
+    });
+  };
+
+  /**
+   * TODO:
+   * Router 수정 후 url 재수정해야합니다.
+   */
+  const movePageBySearch = (value: string) => {
+    const newHistory = history.length === 10 ? [...history].slice(0, 9) : [...history];
+    setHistory([{ id: nanoid(), content: value, day: getDateFormat('', 'dot') }, ...newHistory]);
+    setNameForSearch('');
+    setShowWordList(false);
+    RouterHistory.push(`/search`);
+    // RouterHistory.push(`/search?item=${value}`);
+  };
+
+  const handleClickImg = () => {
+    checkLength(nameForSearch) && movePageBySearch(nameForSearch);
+  };
+
+  const checkLength = (value: string): boolean => {
+    if (value.length === 0) {
+      setIsOccuredError(true);
+      setShowWordList(false);
+
+      setTimeout(() => {
+        setIsOccuredError(false);
+      }, timeToShowError);
+      return false;
+    }
+    return true;
+  };
+
+  const changeSearchName = (pos: 'ArrowUp' | 'ArrowDown') => {
+    if (recommendWords.length) {
+      let newIdx: number = 0;
+      if (pos === 'ArrowUp')
+        if (idxForChoicedWord === 0 || idxForChoicedWord === -1) newIdx = recommendWords.length - 1;
+        else newIdx = idxForChoicedWord - 1;
+      else if (pos === 'ArrowDown')
+        newIdx = idxForChoicedWord === recommendWords.length - 1 ? 0 : idxForChoicedWord + 1;
+      setNameForSearch(recommendWords[newIdx]);
+      setIdxForChoicedWord(newIdx);
+    }
+  };
+
+  const handleKeyUpInput = (e: React.KeyboardEvent<HTMLInputElement>) => {
+    if (e.code === 'Enter') return;
+    if (e.code === 'ArrowUp') {
+      changeSearchName('ArrowUp');
+      return;
+    }
+    if (e.code === 'ArrowDown') {
+      changeSearchName('ArrowDown');
+      return;
+    }
+
+    const matchedWords = words
+      .filter((word) => word.search(getRegExp(e.currentTarget.value)) !== -1)
+      .sort((a, b) => a.length - b.length)
+      .slice(0, 10);
+
+    setRecommendWords(matchedWords);
+    setIdxForChoicedWord(-1);
+  };
+
+  const handleKeyPressInput = (e: React.KeyboardEvent<HTMLInputElement>) => {
+    if (e.code === 'Enter') {
+      checkLength(e.currentTarget.value) && movePageBySearch(e.currentTarget.value);
+      e.currentTarget.blur();
+    }
+  };
+
+  const handleChangeInput = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setNameForSearch(e.currentTarget.value);
+  };
+
+  return (
+    <SearchContainer id="search">
+      <SearchInput
+        onFocus={handleFocusInput}
+        placeholder="검색어를 입력해 주세요"
+        onKeyUp={handleKeyUpInput}
+        onKeyPress={handleKeyPressInput}
+        onChange={handleChangeInput}
+        value={nameForSearch}
+        maxLength={15}
+      />
+      <Button>
+        <SearchImg src="images/search.png" onClick={handleClickImg} />
+      </Button>
+
+      {showWordList && (
+        <WordList
+          handleClick={handleClick}
```

Comment:
props로 전달된 이벤트이름은 on~~ 으로 하는 건 어떨까요?
```suggestion
          onHistoryClick={handleClickHistory}
```
같은 식으로요  😁

Reply:
앗 저번에 말씀드린 부분이긴한데 초반에 handle로 다 통일시키기로 해서,,, 이미 너무 많은것들이 handle로 이루어져있어서 지금 바꾸기엔 타이밍이 애매하네유,,,